pkg/ast/structs.go

```diff
@@ -81,8 +81,10 @@ type ParseError struct {
 }
 
 type TimeModifier struct {
-	RelativeTime RelativeTimeModifier
-	AbsoluteTime string
+	RelativeTime   RelativeTimeModifier
+	AbsoluteTime   string
+	ChainedOffsets []RelativeTimeOffset
+	ChainedSnaps   []string
```

Comment:
Seems like this is only used for Calculating the `startEpoch` and `endEpoch` and there is a `ChainedRelativeTimestamp` struct below. 

I would suggest to refactor the code so that you replace these two with `chainedRelativeTimestamp: *ChainedRelativeTimestamp ` or you remove the `ChainedRelativeTimestamp` completely and keep these two `fields` and return `ast.TimeModifier` for `ChainedRelativeTimestamp` definition instead of `ast.ChainedRelativeTimestamp`.